backend/games/word_guesser/WordGuesser.py

```diff
@@ -0,0 +1,124 @@
+from enum import IntEnum
+from random import randint
+from typing import Dict, List
+
+MAX_GUESSES = 6
+
+# initialize valid guesses and possible solutions
+SOLUTIONS: str
+VALID_WORDS: str
+
+try:
+    solutions_file = open(
+        "backend/games/word_guesser/WordGuesser_Solutions.txt"
+    )
+    guesses_file = open(
+        "backend/games/word_guesser/WordGuesser_ValidGuesses.txt"
+    )
+    SOLUTIONS = solutions_file.read()
+    SOLUTION_LIST = SOLUTIONS.splitlines()
+
+    VALID_WORDS = guesses_file.read()
+    VALID_WORDS_LIST = VALID_WORDS.splitlines()
+except:
+    print("Files not readable. Check file or file path.")
+
+
+class CorrectnessScore(IntEnum):
+    correct = 3
+    in_word = 2
+    not_in_word = 1
+
+
+class WordGuesser:
+    def __init__(self, goal=None):
+        self.guesses = {"guess": "", "results": []}
+        self.game_board: List[str] = []
+        self.progress: Dict = {}
+        self.goal_word = goal
+        self.guess_count = 0
+
+    def clear_data(self):
+        self.guesses = {"guess": "", "results": []}
+        self.game_board: List[str] = []
+        self.progress: Dict = {}
+        self.goal_word = ""
+        self.guess_count = 0
+
+    def start_game(self):
+        self.clear_data()
+
+        new_word_index = randint(0, len(SOLUTION_LIST) - 1)  # nosec
+        self.goal_word = SOLUTION_LIST[new_word_index]
+
+    def end_game(self) -> str:
+        output = self.check_win()
+        if output == True:
+            return "You guessed the word!! Nice Job"
+        else:
+            return "You failed to guess the word in 6 guesses. better luck next time."
+
+    # holds game history
+    def track_progress(self) -> Dict:
+        tmp_keys = ""
+        tmp_results = []
+        for k, v in self.guesses.items():
+            if "guess" in k:
+                tmp_keys = v
+            if "results" in k:
+                tmp_results = v
+            self.progress[tmp_keys] = tmp_results
+        return self.progress
+
+    def check_win(self) -> bool:
+        idx = len(self.game_board) - 1
+        if self.game_board[idx] == self.goal_word:
+            return True
+        elif self.guess_count > 6:
+            return False
+        return False
+
+    # performs all logic checks on a word to determine if it is valid
+    def validate_guess(self, guess) -> bool:
+        if len(guess) != 5:
+            return False
+        if not guess.isalpha():
+            return False
+        if guess not in VALID_WORDS:
```

Comment:
You can make `VALID_WORDS` a set for faster lookup here. And you can keep `SOLUTION_WORDS` as a list for random selection. Each should the right data structure for its purpose.